Formula/petsc-complex.rb

```diff
@@ -0,0 +1,49 @@
+class PetscComplex < Formula
+  desc "Portable, Extensible Toolkit for Scientific Computation"
+  homepage "https://www.mcs.anl.gov/petsc/"
+  url "http://ftp.mcs.anl.gov/pub/petsc/release-snapshots/petsc-lite-3.9.3.tar.gz"
+  sha256 "8828fe1221f038d78a8eee3325cdb22ad1055a2f0671871815ee9f47365f93bb"
+
+  bottle do
+    sha256 "733080c80cec9ad2c7da2fb5d922f94889421779eac6c291c9a62152f333fb6b" => :high_sierra
+    sha256 "e8858a1cec95cc42417773aac1b7e85db1c4938e149f7b27a20ad739a51cb395" => :sierra
+    sha256 "4b8ddec36720b2778df1fcdcf765bcf7a847abbeec6a7df9f09067b4ca4fb505" => :el_capitan
+  end
+
+  depends_on "hdf5"
+  depends_on "hwloc"
+  depends_on "metis"
+  depends_on "netcdf"
+  depends_on "open-mpi"
+  depends_on "scalapack"
+  depends_on "suite-sparse"
+  conflicts_with "petsc", :because => "petsc must be installed with either real or complex support, not both"
+
+  def install
+    ENV["CC"] = "mpicc"
+    ENV["CXX"] = "mpicxx"
+    ENV["F77"] = "mpif77"
+    ENV["FC"] = "mpif90"
+
+    args = %W[
+      --prefix=#{prefix}
+      --with-debugging=0
+      --with-x=0
+    ]
+    system "./configure", "--with-scalar-type=complex", *args
+    system "make", "all"
+    system "make", "install"
+  end
+
+  test do
+    test_case = "#{share}/petsc/examples/ksp/ksp/examples/tutorials/ex1.c"
```

Comment:
Use `pkgshare`

Reply:
@izaid said:
> For the petsc-complex formula specifically, the pkgshare variable can't be used because petsc-complex is installed with the same directory as petsc. This is important so that the C include directory remains the same, regardless of whether one is using petsc or petsc-complex. That's why the share variable is used there.

But you're setting `--prefix=#{prefix}`. Am I missing something? Doesn't petsc go into e.g., `/usr/local/Cellar/petsc` and petsc-complex go into `/usr/local/Cellar/petsc-complex`?

Reply:
I just printed out the variables. Their values are as follows

`prefix = /usr/local/Cellar/petsc-complex/3.9.3`
`share = /usr/local/Cellar/petsc-complex/3.9.3/share`
 `pkgshare = /usr/local/Cellar/petsc-complex/3.9.3/share/petsc-complex`

So `prefix` and `share` are fine. However, `petsc` puts things into `/usr/local/Cellar/petsc-complex/3.9.3/share/petsc`. Consider that for include, i.e. `/usr/local/Cellar/petsc-complex/3.9.3/include/petsc`, this makes sense, so that the C include directory doesn't change depending upon whether or not `petsc` or `petsc-complex` is installed.